examples/raw/download.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/**
+ * Minimal example to download a file
+ * Assumes the last part of the path is the file name
+ *
+ * Try it using the built-in webserver
+ * `ACCESS=youraccesstring php -S 0.0.0.0:8080`
+ * then visit http://localhost:8080/download.php
+ */
+
+use Storj\Uplink\Exception\Object\ObjectNotFound;
+use Storj\Uplink\ListObjectsOptions;
+use Storj\Uplink\Uplink;
+
+ini_set('display_errors', 1);
+
+require '../../vendor/autoload.php';
+
+$project = Uplink::create()->parseAccess(getenv('ACCESS'))->openProject();
+$project->ensureBucket('raw');
+
+$filename = trim($_SERVER['PATH_INFO'] ?? '', '/');
+if ($filename) {
+    try {
+        $download = $project->downloadObject('raw', $filename);
+    } catch (ObjectNotFound $e) {
+        http_response_code(404);
+        echo "<p>File not found</p>";
+        return;
+    }
+
+    $type = $download->info()->getCustomMetadata()['Content-Type'];
+    header("Content-Type: $type");
+    $download->readIntoResource(fopen('php://output', 'w'));
+    return;
+}
+
+$nFiles = 0;
+$listHtml = '<ul>';
+foreach ($project->listObjects('raw', new ListObjectsOptions('', '', false, true, true)) as $objectInfo) {
```

Comment:
Can we make all 'options' parameters more declarative?

For the Java bindings we made options params to be varargs. Each option can be created with a static method of `ObjectListOptions`: https://github.com/storj/uplink-java/blob/e016f10d2f35f94711629de914bd46e61b1f9449/src/main/java/io/storj/ObjectListOption.java

This way we could have a clearer code like:

```
$project->listObjects('raw', ListObjectsOptions::system(), ListObjectsOptions::custom())
```
instead of:
```
$project->listObjects('raw', new ListObjectsOptions('', '', false, true, true))
```

Reply:
The Jave solution seems a bit overengineerd. Also PHP does not come with typesafe enums out-of-the-box.

I have removed the constructor and changed it it to with'ers

```php
$objects = $project->listObjects(
    'raw', 
    (new ListObjectsOptions())
        ->withSystemMetadata()
        ->withCustomMetadata()
);
```

Reply:
Also for DownloadOptions. 

I'm not sure if I should keep the `with` method prefix. 

It's idiomatic, but also to prevent a conflict with the getter. But we don't have any getters.